### ISSUE TITLE
Cusom dataset with one class

### ISSUE THREAD

**Issue by watermellon2018** [yassouali/CCT#53]
How i can use this model for custom dataset with one class? I get error, i change cross entropy loss to binary loss but get other error. Can you help me?

**Comment by yassouali** [comment]
Hi @watermellon2018 

yes if you only have a foreground class (i think in this case, you have two classes, background and foreground), with 0/1, you can either use a BCE with one output and a threshold of say 0.5 when doing inference to choose between FG and GB. Or simply use two class output and train your model with softmax.

**Comment by tiberium24** [comment]
Hello @yassouali. I am trying the same. Can you elaborate a bit on how to do that?
I have two classes 0 and 1 and when I am training with CE loss, it only learns the 0 class. No imbalance or other issue.
Not sure how to apply BCE loss or why it would be needed.
